Mobile and web tracking recommendations

agnessnowplow opened the pull request:
This PR has multiple aims:
1.  to **bridge the gap between tracking and modelling**, especially when it comes to using the **unified model**: what contexts to use in which case and how to enable them. I already added a new table under tracker protocol for this. I think we need separate pages that focus on the individual schemas (unless they exist somewhere?) and provide the link just like I already did with the enrichments. [Deploy preview](https://deploy-preview-642--snowplow-docs.netlify.app/docs/collecting-data/collecting-from-own-applications/out-of-the-box-tracking)
2. to help understand and follow where we are with **moving away from the canonical event models** (for us as well!) For that what I thought we could do is add `Web` and `Mobile` fields for each table depicted in the canonical event page where we either tick that it is available and recommended to be used from there directly still OR if there are currently multiple choices (e.g. for mobile it is rather recommended to be taken from a context) we explain it there. I have not done much work on this yet as it is bigger and I want some opinions first.[ Deploy preview](https://deploy-preview-642--snowplow-docs.netlify.app/docs/understanding-your-pipeline/canonical-event/#canonical-event-model)
3. to help onboarding / prospects understand **what fields are tracked out of the box across mobile** trackers not just for web as currently depicted [here](https://docs.snowplow.io/docs/understanding-your-pipeline/canonical-event/) I think if we complete point 2 this will solve this immediately.

mscwilson commented:
Re 1:
Great idea to add clarity here. I think we need another column in the table for "how to enable it in the trackers/pipeline". e.g. "autotracked by default for web trackers" "must be manually tracked" etc.
This is web **or** mobile tracking yeah not just web **&** mobile? They don't need to be having events from both places to use the unified model?
Would be good to specify which trackers we're referring to here for web/mobile.

Re 2:
In general the tables on this page aren't the clearest, could really benefit from a key. `Req'd` for what? `Source` - what's the difference between Pipeline and Default Enrichment? The deprecated fields could be more clearly marked too, like in the first column with their name (crossed out?), not in the last column.
Anyway, what does/would it mean if the row has a tick or cross under web or mobile?
An example default web and default mobile event would be great for this page/section.

mscwilson commented on docs/collecting-data/collecting-from-own-applications/snowplow-tracker-protocol/recommended-tracking-design-web-mobile/index.md:
```diff
@@ -0,0 +1,34 @@
+
+# Recommended Tracking Design for Web & Mobile Events
+
+Depending on your business needs you may decide to come up with your own custom contexts / self-describing events. However, for out-of-the-box solutions Snowplow designed a set of contexts specifically useful for web and mobile tracking which we used as a base for our data models, too. 
```
`context entities` here and elsewhere in the intro

rlh1994 replied:
I think we've actually moved to ideally just calling them entities everywhere (https://deploy-preview-642--snowplow-docs.netlify.app/docs/understanding-your-pipeline/entities/)

agnessnowplow commented:
**Re1:** Thanks for the feedback! I added a new `how to enable it in the trackers/pipeline` column as I had this in the past but thought the link in the first column should cover it altogether but I agree an upfront clarification is still useful.

Changed the `&` to `or` in the title to make it clearer but yes, it could be one of the three options: web only / mobile only / both. The main usage at first should be using both at the same time as we have the solidified platform oriented models but it was designed with the intention to potentially shift usage into one model that can do all in one. It already works like that so everything is optional.

I will add one more field in the intro about which trackers we are covering here, good point!

**Re2:** I was hesitant to change much as I thought it is another team that maintains that page but then I saw Ryan making the last changes on the colums you asked about (perhaps some styling?) so will ask as I have no idea what the `Reqd` column means for instance.

I tried crossing out the deprecated field names and I think it makes it much more obvious, good shout!

The cross and tick mark was meant to label whether that field comes out of the box for that platform (I was thinking of web only fields being null for mobile?) It is just to give reassurance to anyone looking at the screen what is relevant for which platform tracking as it was originally only written for web which made people ask about the same page for the mobile equivalent. Perhaps we can make that obvious in an explanatory sentence in the intro.

miike commented:
* `geolocation` should be a ticket for web (using the geolocation plugin).
* For mobile do we need to explicitly call out React Native if it doesn't have the same support?
* I wouldn't mind moving this one level higher in the hierarchy of pages (outside of tracker protocol)
* mobile has some consent events / contexts (though not sure about consent_preferences specifically)
* web also has `appliation_error` using the error-tracking plugin
* I'd remove ua_context as we consider it deprecated
* IAB context should be ticked for both (as it's an enrichment rather than collected by the tracker)
* web also supports session context (`client_session`)

matus-tomlein commented on docs/collecting-data/collecting-from-own-applications/snowplow-tracker-protocol/recommended-tracking-design-web-mobile/index.md:
```diff
@@ -0,0 +1,34 @@
+
+| Context               | Web | Mobile | Recommendation                                                           | How to enable it in the trackers/pipeline | How to enable it in the unified model                      | Field name                                                  |
+|-----------------------|-----|--------|--------------------------------------------------------------------------|-------------------------------------------|------------------------------------------------------------|-------------------------------------------------------------|
+| page view context     | ✅   | ❌      | required for web (needed for page_view_id)                               | autotracked by default for web trackers   | snowplow__enable_web: true                                 | com_snowplowanalytics_snowplow_web_page_1                   |
```
Could we link to more details about the individual context entities? Probably the easiest would be to link to the schema on iglu central, for some of them, we may also link to the tracking docs configuration, not sure if that would be preferable.

matus-tomlein commented on docs/collecting-data/collecting-from-own-applications/snowplow-tracker-protocol/recommended-tracking-design-web-mobile/index.md:
```diff
@@ -0,0 +1,34 @@
+| Context               | Web | Mobile | Recommendation                                                           | How to enable it in the trackers/pipeline | How to enable it in the unified model                      | Field name                                                  |
+|-----------------------|-----|--------|--------------------------------------------------------------------------|-------------------------------------------|------------------------------------------------------------|-------------------------------------------------------------|
+| page view context     | ✅   | ❌      | required for web (needed for page_view_id)                               | autotracked by default for web trackers   | snowplow__enable_web: true                                 | com_snowplowanalytics_snowplow_web_page_1                   |
+| screen view events    | ❌   | ✅      | required for mobile (needed for screen_view_id)                          | must be manually tracked                  | snowplow__enable_mobile: true                              | com_snowplowanalytics_mobile_screen_view_1                  |
```
These are events that don't really fit under the Context table. Maybe we can create a separate table for OOTB events? That would also including background, foreground, install, deep link, link tracking and other events?

matus-tomlein reviewed (commented):
As a suggestion, would it make sense to have short explanations under the table for the different context entities and events? I mean from the table we could link to lower parts of the page, where each event/entity would be explained. I can imagine that for someone not familiar with these concepts, just listing them might not be enough.

agnessnowplow commented on docs/collecting-data/collecting-from-own-applications/out-of-the-box-tracking/index.md:
```diff
@@ -0,0 +1,146 @@
+| app errors [(schema)](https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow/application_error/jsonschema/1-0-2)           | ✅    |  ✅      | web: error tracking plugin                                                                        | ?                                         | snowplow__enable_app_errors_module: true (optional module) |  com_snowplow ...
+
+Below you will find more detailed information on the individual Snowplow events and context entities:
+
```
It is not ready yet. :) I'm putting in some canvas for Matus and other tracking specialists to fill out basically is where we are at.

agnessnowplow replied:
Also toying with the layout ideas, not sure if the example method will be beneficial without desciption on what they actually are. Not sure if it is useful as is.

agnessnowplow commented on docs/collecting-data/collecting-from-own-applications/out-of-the-box-tracking/index.md:
yeah I was thinking this would have to be duplicated in the relevant modeling docs too. I think I will just remove it but add a link to it here as I still think it would be beneficial to link the concepts together somehow.

agnessnowplow commented on docs/collecting-data/collecting-from-own-applications/out-of-the-box-tracking/index.md:
```diff
@@ -0,0 +1,146 @@
+| screen view events [(schema)](https://github.com/snowplow/iglu-central/tree/master/schemas/com.snowplowanalytics.snowplow/screen_view/jsonschema)   | ❌   | ✅      | required for mobile (needed for screen_view_id)                          | must be manually tracked                  | snowplow__enable_mobile: true                              | com_snowplowanalytics_mobile_screen_view_1           ...
+## Context entities
+
+| context               | web | mobile | recommendation                                                           | how to enable it in the trackers / pipeline | how to enable it in the unified model                      | field name                                                  |
+|-----------------------|-----|--------|--------------------------------------------------------------------------|-------------------------------------------|------------------------------------------------------------|-------------------------------------------------------------|
```
would it be clearer to add those "pre-" and "suffixes" in brackets maybe? it's a bit of a nightmare to update this table with all this markdown having to use tools to translate back and forth each time there is a change but I guess once solidified we could add tabs to support different warehouse outputs if otherwise doesn't look right

rlh1994 replied:
Maybe just a note on the page somewhere is probably good enough

agnessnowplow commented:
Closing this PR as we now have the new one here: https://github.com/snowplow/documentation/pull/650
